app/src/main/java/com/davidtomas/taskyapp/features/auth/presentation/register/RegisterAction.kt

```diff
@@ -0,0 +1,11 @@
+package com.davidtomas.taskyapp.features.auth.presentation.register
+
+sealed class RegisterAction
+data class OnUserNameChanged(val userName: String) : RegisterAction()
+data object OnUserNameFocusChanged : RegisterAction()
+data class OnEmailChanged(val email: String) : RegisterAction()
+data object OnEmailFocusChanged : RegisterAction()
+data class OnPasswordChanged(val password: String) : RegisterAction()
+data object OnChangePasswordVisibility : RegisterAction()
+data object NavigateBackToLogin : RegisterAction()
+data object OnRegisterButtonClicked : RegisterAction()
```

Comment:
I'd recommend to usually put members of a sealed class inside the class body. Otherwise, you'll always need to manually look which subclasses there are for `RegisterAction`. Making them part of the class, you can just type `RegisterAction.` and the IDE suggestions will tell you all subclasses